ONSdigital/eq-survey-runner-benchmark
# Add session/download-pdf endpoints and add graph to represent values

### What is the context of this PR?
To update the benchmark outputs in order for us to monitor and highlight the performance of more intensive endpoints by including `session` and `download-pdf`. 

`benchmark_stats` has been provisioned with the 99th percentiles for the `session` and `download-pdf` endpoints. In order to achieve this, the happy path has been updated to include `download-pdf` such that the csv file created when running the benchmark contains the percentiles for this endpoint. The output "string" of the benchmark stats has been updated to include the new metrics for both Github and non-Github outputs.

`visualise_results` has been updated to support the new metrics. Currently, as the 99th percentiles for both `session` and `download-pdf` are significantly larger than the original figures, it was decided that outputting 2 graphs was the most appropriate solution. Thus, 2 dataframes have been created, one for the original percentiles and a new one for the additional metrics. The "image" of the graphs created use subplots to place both graphs created next to each other.

Various tests have been updated and parametrised. In addition to this `conftest` was updated to include more fixtures that support the new endpoints.

### How to review
- Check if the changes make sense
- Suggestions for refactoring
- Ensure all tests pass
- Use the [README](https://github.com/ONSdigital/eq-survey-runner-benchmark?tab=readme-ov-file#visualising-benchmark-results) to test the updated `visualise_results` script
- Spin up a benchmark pipeline and test the output to Slack and spin up a Dependabot staging pipeline to test the output to Github
